### PR TITLE
feat(ui5-time-picker): introduce `open` property

### PR DESCRIPTION
**Pull request by hinzzx**
Add `open` property to replace the `openPicker` method in the `ui5-time-picker` component.

BREAKING CHANGE: Removed `openPicker()`, `closePicker()`, and `isOpen()` methods.

Before, the `ui5-time-picker` popover could be opened and closed by calling `openPicker()` and `closePicker()`:
```ts
const timePicker = document.getElementById("exampleID");
timePicker.openPicker();
timePicker.closePicker();
```

Now, the popover is opened and closed by setting the `open` property to true or false:
```ts
const timePicker = document.getElementById("exampleID");
timePicker.open = true;
timePicker.open = false;
```

Related to: https://github.com/SAP/ui5-webcomponents/issues/8461

### CODE REVIEW

File: packages/main/src/TimePicker.ts [hunk 1]
```diff
@@ -363,56 +371,21 @@ class TimePicker extends UI5Element implements IFormInputElement {
...
 	}
 
 	onResponsivePopoverAfterClose() {
-		this._isPickerOpen = false;
+		this.open = false;
 	}
 
 	onResponsivePopoverAfterOpen() {
-		this._isPickerOpen = true;
+		this.open = true;
```
Comment (didip1000) [inline packages/main/src/TimePicker.ts:388]: Is this necessary?

File: packages/main/src/TimePicker.ts [hunk 2]
```diff
@@ -558,15 +531,15 @@ class TimePicker extends UI5Element implements IFormInputElement {
...
 			e.preventDefault();
-			this.togglePicker();
+			this._togglePicker();
 		}
 
 		const target = e.target as Node;
 
-		if ((this._getInput().isEqualNode(target) && this.isOpen()) && (isTabNext(e) || isTabPrevious(e) || isF6Next(e) || isF6Previous(e))) {
-			this.closePicker();
+		if ((this._getInput().isEqualNode(target) && this.open) && (isTabNext(e) || isTabPrevious(e) || isF6Next(e) || isF6Previous(e))) {
```
Comment (didip1000) [inline packages/main/src/TimePicker.ts:539]: Comparison by ID's would be cheaper like in DatePicker

Comment (hinzzx) [reply packages/main/src/TimePicker.ts:539]: Yeah I thought of that too, but got mistaken that in the DatePicker we compare by **_id**, while we actually use **.id**.
Will do, thanks

File: packages/main/src/TimePicker.ts (hunk 1 above)
Comment (tsanislavgatev) [reply packages/main/src/TimePicker.ts:388]: After a discussion with Stoyan, we agreed that setting an attribute/property directly to the popover isn't a valid use case, as we are the one creating and managing the popover, that's why we don't need to 'care' if someone else changes the state. Also the logic on timepicker side should only be: set open to false -> close, set open to true -> open. So both handlers will be deleted, from TS and hbs files.

**Review by tsanislavgatev — CHANGES_REQUESTED** [review]

**Review by tsanislavgatev — APPROVED** [review]

**Review by nnaydenow — APPROVED** [review]

**Pull request merged**